src/AlternativesPerformance.cpp

```diff
@@ -0,0 +1,142 @@
+#include "AlternativesPerformance.h"
+#include <typeinfo>
+
+AlternativesPerformance::AlternativesPerformance(
+    std::vector<Performance> &perf_vect,
+    std::map<std::string, std::string> alt_assignment)
+    : PerformanceTable(perf_vect) {
+  if (mode_ != "alt") {
+    throw std::domain_error(
+        "Performance table mode should be alt to assign default categories.");
+  }
+  if (alt_assignment == std::map<std::string, std::string>{{"", ""}}) {
+    for (std::vector<Perf> pv : pt_) {
+      std::string altName = pv[0].getName();
+      alt_assignment_[altName] = "";
+    }
```

Comment:
If the default alt_assignment is an empty hash map, do we plan on leaving it that way or do we want to assign it random categories ? idk

Reply:
This discussion would have consequences on other constructors.

Reply:
In a sense it's useless to use an AP object if we don't use categories as it would just be a performance table. But does it make sense to have randomly assigned categories? I would need to think that through a bit more but given the thesis we can't really assign categories randomly and stay consistent right?

Reply:
Maybe the AP object can be created with blank categories in the main model function and the categories can be assigned afterwards, but sure this is kind of useless to have it like that. Anyway the default map value might never be used, not sure this means we should remove it

Reply:
It's fine keeping it that way, maybe in the doc we could specify that we should not use an AP with the default empty categories and use a pt instead!

Reply:
Ok I'll add that